### PR TITLE
Space Travelers' Hub- Group Capstone Project

### PR DESCRIPTION
**Pull request by meryemsanem**
### The Space Travelers' Hub consists of Rockets, Missions, and the My Profile section.

- The **Rockets** section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking. 

- The **Missions** section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.

- The **My Profile** section displays all reserved rockets and space missions.
-------------------------------------------------------------------------------------------------------------------------------------------

- [ ] Create React App.
- [ ] Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/react_router.md).

- [ ] Create routes and view components: rockets, missions, my profile. 
- [ ] Use <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
- [ ] Create directories for all Redux state slice files: rockets, missions.
- [ ] Redux: Fetch data and update Redux store
- [ ] Upon first render fetch data from the SpaceX API endpoints:
- Rockets: https://api.spacexdata.com/v4/rockets
- Missions: https://api.spacexdata.com/v3/missions
- [ ] Once the data are fetched, dispatch an action to store the selected data in Redux store:
- Rockets:
1. id
2. rocket_name
3. description
4. flickr_images

- Missions:
1. mission_id
2. mission_name
3. description

- [ ] Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes.
- [ ] When a user clicks the "Reserve rocket" button action needs to be dispatched to update the store. You need to get the ID of the reserved rocket and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all rockets, but the selected rocket will have an extra key reserved with its value set to true. You could use a JS filter() or map() to set the value of the new state.
- [ ] Create a reducer and action dispatcher for the "Join Mission" button.
- [ ] Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design) .
- [ ] Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
- [ ] Render a list of all joined missions (use filter()).
- [ ] Render a list of all reserved rockets (use filter()).
- [ ] Add test for all your components and functions.

### CODE REVIEW

File: src/__tests__/Rockets.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  });
+
+  const { asFragment } = render(
+    <Provider store={store}>
+      <Rockets />
+    </Provider>,
+  );
+
+  expect(asFragment()).toMatchSnapshot();
+});
```
Comment (danmainah) [inline src/__tests__/Rockets.test.js:39]: - [ ] @meryemsanem and @mahdinoori2000 Great work adding the tests for the components, however, you missing tests for the App component, also consider adding tests for your pure functions in the redux directory.

File: src/pages/Rockets.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+              {rocket.description}
+            </p>
+            <button
+              type="button"
+              className="rocket-btn"
+              data-testid={`cancel-reservation-button-${rocket.id}`}
+              onClick={() => handleCancelReservationButtonClick(rocket.id)}
+            >
+              Cancel Reservation
+            </button>
```
Comment (danmainah) [inline src/pages/Rockets.js:55]: - [ ] @Team great work displaying rockets, however, the design of the Cancel Reservation  button doesn't match the design given, check the comparison below

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image |  image

File: src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          <NavLink to="/" className="nav-link">Rockets</NavLink>
+        </li>
+        <li>
+          <NavLink to="/missions" className="icon nav-link">Missions</NavLink>
+        </li>
+        <li>
+          <NavLink to="/profile" className="nav-link">My Profile</NavLink>
+        </li>
+      </ul>
+    </nav>
```
Comment (danmainah) [inline src/components/Navbar.js:24]: - [ ] @Team great work on your navbar however, there is a slight difference between your link's color and the color given in the design, the same applies to the reserve rockets button, check the comparison below for better understanding:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image  |  image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @meryemsanem and @mahdinoori2000 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added navigation and ensured the user can access all pages
✔️  Ensured rockets and missions are fetched and displayed correctly.
✔️  Ensured users are able to join and leave a mission, reserve and cancel reservations for rockets
✔️ Added tests for components and ensure all tests are passing.




# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Strangeal — APPROVED** [review]
Hi @meryemsanem and @mahdinoori2000, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**